src/chart/generateCategoricalChart.tsx

```diff
@@ -238,6 +239,21 @@ const getTooltipContent = (
       // graphic child has data props
       const entries = data === undefined ? displayedData : data;
       payload = findEntryInArray(entries, tooltipAxis.dataKey, activeLabel);
+    } else if (tooltipAxis.dataKey && tooltipAxis.allowSelectNearestValue && tooltipAxis?.categoricalDomain) {
```

Comment:
nit: you destructure `dataKey` above but use `tooltipAxis.dataKey` here. Could maybe use destructured values for all three checks here?